src/main/scala/beam/agentsim/infrastructure/ChargingNetwork.scala

```diff
@@ -59,7 +59,8 @@ class ChargingNetwork(managerId: Id[VehicleManager], chargingStationsQTree: Quad
     * @param vehicleId vehicle Id
     * @return charging vehicle
     */
-  def lookupVehicle(vehicleId: Id[BeamVehicle]): Option[ChargingVehicle] = vehicles.get(vehicleId)
+  def lookupVehicle(vehicleId: Id[BeamVehicle]): Option[ChargingVehicle] =
+    chargingZoneKeyToChargingStationMap.values.view.flatMap(_.lookupVehicle(vehicleId)).headOption
```

Comment:
This seems less performant, and a logic change - was this meant to be in this PR?

Reply:
This is the main point of the PR. Generally, it does the same as in the previous version. But instead of building a Map (this operation is O(n)) and get the vehicle by Id, it looks up the vehicle in the internal map and if not found it looks up the vehicle in the PriorityQueue.

Reply:
Oy, this got complicated the more I looked into it. It seems like it is immutable at first, and I was going to suggest making the map a val - however I see that it is not - so I understand the change better.